dashboard/app/models/school_stats_by_year.rb

```diff
@@ -42,5 +42,23 @@
 
 class SchoolStatsByYear < ActiveRecord::Base
   self.primary_keys = :school_id, :school_year
+
   belongs_to :school
+
+  # Loads/merges the data from a CSV into the table.
+  # Requires a block to parse the row.
+  # @param filename [String] The CSV file name.
+  # @param options [Hash] Optional, the CSV file parsing options.
+  def self.merge_from_csv(filename, options = {col_sep: "\t", headers: true, quote_char: "\x00"})
+    CSV.read(filename, options).each do |row|
+      parsed = yield row
+      loaded = find_by(primary_keys.map(&:to_sym).map {|k| [k, parsed[k]]}.to_h)
+      if loaded.nil?
+        SchoolStatsByYear.new(parsed).save!
+      else
+        loaded.assign_attributes(parsed)
+        loaded.update!(parsed) if loaded.changed?
+      end
+    end
+  end
```

Comment:
- This method is practically identical to ones in other models, it should be refactored into a helper so the same code can be used in multiple models.
- This logic is complicated enough that it would benefit from unit tests making it clear what exactly it's doing and that it works as expected (without relying on manual one-off testing for this).
- The use of `find_by` is probably not the most efficient (though this doesn't matter for a one-off import, it's something we might want to return to later if this code will be re-used often).
- The default `options` are equivalent to the `CSV_IMPORT_OPTIONS` in other models, I'd suggest referencing a shared constant somewhere for these as well.